sdc/crypto/token_helper.py

```diff
@@ -0,0 +1,105 @@
+import json
+
+from jwcrypto import jwe, jwt
+from jwcrypto.common import base64url_decode
+from jwcrypto.jwe import InvalidJWEData
+from jwcrypto.jws import InvalidJWSSignature, InvalidJWSObject
+from jwcrypto.jwt import JWTInvalidClaimFormat, JWTMissingClaim, JWTExpired
+
+from sdc.crypto.invalid_token_exception import InvalidTokenException
+
+
+def decrypt_jwe(encrypted_token, secret_store, purpose, default_kid=None):
+    try:
+        jwe_token = jwe.JWE(algs=['RSA-OAEP', 'A256GCM'])
+        jwe_token.deserialize(encrypted_token)
+
+        try:
+            jwe_kid = extract_kid_from_header(encrypted_token)
```

Comment:
This has changed slightly in a new PR where the default is removed and i have added logging of the KID being used. the Logging will be really useful when rotating keys and identifying keys that are in use.
https://github.com/ONSdigital/eq-survey-runner/pull/1183/files#diff-9c05c59fdd2f51a0551137834f6ae523R22